mixnet/node/src/lib.rs

```diff
@@ -178,12 +232,54 @@ impl MixnetNode {
 
     async fn forward(
         pool: &ConnectionPool,
+        max_retries: usize,
+        retry_delay: Duration,
         body: Body,
         to: NymNodeRoutingAddress,
     ) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
         let addr = SocketAddr::try_from(to)?;
-        body.write(&mut *pool.get_or_init(&addr).await?.lock().await)
-            .await?;
+        if body
+            .write(&mut *pool.get_or_init(&addr).await?.lock().await)
+            .await
+            .is_err()
+        {
+            let pool = pool.clone();
+            tokio::spawn(retry(addr, pool, max_retries, retry_delay, body));
+        }
         Ok(())
     }
 }
+
+async fn retry(
```

Comment:
What do you think about reusing the `retry_backoff` function that you already defined in the `mixnet/client/src/sender.rs`? I didn't check if it's possible, but it seems we don't need to have two different implementations.